packages/core/src/index.ts

```diff
@@ -17,6 +17,18 @@ interface Properties {
   [key: string]: string;
 }
 
+class UIError extends Error {
```

Comment:
actually, if we are not instantiating this class (so, we are not doing any `new UIError(...)`), do we need it?

if it is only about having the type, could be possible to set something like

```
interface UIError extends Error {
  componentStack: string;
}
```